schemas/clab.schema.json

```diff
@@ -303,48 +303,96 @@
                     "$ref": "#/definitions/certificate-config"
                 }
             },
-            "if": {
-                "properties": {
-                    "kind": {
-                        "const": "srl"
+            "allOf": [
+                {
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "description": "type of a node",
+                            "markdownDescription": "node [type](https://containerlab.dev/manual/nodes/#type)"
+                        }
                     }
-                }
-            },
-            "then": {
-                "properties": {
-                    "type": {
-                        "type": "string",
-                        "description": "type of a node",
-                        "markdownDescription": "node [type](https://containerlab.dev/manual/nodes/#type)",
-                        "enum": [
-                            "ixrd1",
-                            "ixrd2",
-                            "ixrd3",
-                            "ixrd2l",
-                            "ixrd3l",
-                            "ixrd4",
-                            "ixrd5",
-                            "ixrd5t",
-                            "ixrh2",
-                            "ixrh3",
-                            "ixrh4",
-                            "ixr6",
-                            "ixr6e",
-                            "ixr10",
-                            "ixr10e"
+                },
+                {
+                    "if": {
+                        "properties": {
+                            "kind": {
+                                "const": "srl"
```

Comment:
Hi @kellerza 
good stuff, since you have a fresh memory of json schema maybe you remember if there is a possibility to have multiple values in the `if` clause?

Ther reason is, that we have more than a single kind constant:

1. `srl` and `nokia_srlinux`
2. `vr-sros` and `vr-nokia_sros`

The latter option is preferrable. But it would be cool if json schema allowed to use multiple consts without code duplication

Reply:
Maybe we could add an anyOf inside the if clause, will give it a go

Reply:
"pattern" seems simple enough

<img width="118" alt="image" src="https://github.com/srl-labs/containerlab/assets/6756881/533d28b3-dfd7-4a5e-a959-408d912133ed">

<img width="125" alt="image" src="https://github.com/srl-labs/containerlab/assets/6756881/b2cfd5fb-c877-4a4e-bb12-904d7d884275">
<img width="144" alt="image" src="https://github.com/srl-labs/containerlab/assets/6756881/747ab5b8-153b-41bc-ad32-347aea85360e">

Reply:
niiice